Azure/sonic-buildimage
# DellEMC S6100: Determine pcie.yaml revision based on firmware version

#### Why I did it

To determine the revision of the pcie.yaml to be used based on BIOS version in DellEMC S6100 platform.

Depends on: https://github.com/Azure/sonic-platform-common/pull/195

#### How I did it

- Added two revisions of pcie.yaml `pcie_1.yaml` and `pcie_2.yaml`
- Included a platform-specific `Pcie` class to provide the revision of the pcie.yaml to be used by pcieutil/pcied.  

#### How to verify it

Execute `pcieutil check` (https://github.com/Azure/sonic-utilities/pull/1672) command and verify the list of PCIe devices displayed.
Logs: [UT_logs.txt](https://github.com/Azure/sonic-buildimage/files/6655450/UT_logs.txt)

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


DellEMC S6100: Determine pcie.yaml revision based on firmware version

#### A picture of a cute animal (not mandatory but encouraged)
